Library/Formula/llvm.rb

```diff
@@ -111,6 +120,37 @@ class Llvm < Formula
   fails_with :gcc
   fails_with :llvm
 
+  # Disable superenv if building with additional runtime libraries (compiler-rt)
+  #
+  # Something about superenv confuses the linker when compiling the support
+  # libraries (compiler-rt). Linking fails with errors similar to:
+  #   ...
+  #   ld: building for iOS simulator, but linking against dylib built for OSX,
+  #     file '/usr/lib/libc++.dylib' for architecture x86_64
+  #   clang: error: linker command failed with exit code 1 (use -v to see invocation)
+  #   make[2]: *** [lib/clang/3.8.0/lib/darwin/
+  #     libclang_rt.ubsan_iossim_dynamic.dylib] Error 1
+  #   ...
+  #   fatal error: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/
+  #     XcodeDefault.xctoolchain/usr/bin/lipo: specifed architecture type (i386)
+  #     for file (/tmp/llvm20150910-69149-1cdqr2s/./lib/clang/3.8.0/lib/darwin/
+  #     libclang_rt.builtins_i386_iossim.a) does not match its cputype
+  #     (16777223) and cpusubtype (3) (should be cputype (7) and cpusubtype (3))
+  #   make[2]: *** [projects/compiler-rt/lib/builtins/
+  #     libclang_rt.builtins_iossim.a] Error 1
+  #   <cleanup and display error>
+  # Unfortunately, the reason the linker is doing this is not known at this
+  # time (11 September 2015). This is not the only incident of its kind; see:
+  #     https://forums.developer.apple.com/thread/13795
+  # so it is possible that this can be removed if/when XCode and/or the CLT get
+  # fixed.
+  #
+  # Builds without these support libraries completed normally, which is why
+  # this change was not required earlier. In addition, the libraries are only
+  # built when clang is built
+
+  env :std if build.with? "clang"
```

Comment:
Hmmm... Thing is that some earlier linker errors complained about building for iOS simulator (i386) but trying to link against libraries for OS X (x86_64). And that worked fine when pointed to the iOS simulator dylibs. I believe those are i386, but I'll check next time I'm at my computer.

I also remember testing unsuccessfully with --universal set once, but don't remember if it was that particular bit that killed the build. I'll check when I can

> On Sep 19, 2015, at 03:15, Mike McQuaid <notifications@github.com> wrote:
> 
> In Library/Formula/llvm.rb:
> 
> > +  #     libclang_rt.builtins_i386_iossim.a) does not match its cputype
> > +  #     (16777223) and cpusubtype (3) (should be cputype (7) and cpusubtype (3))
> > +  #   make[2]: *** [projects/compiler-rt/lib/builtins/
> > +  #     libclang_rt.builtins_iossim.a] Error 1
> > +  #   <cleanup and display error>
> > +  # Unfortunately, the reason the linker is doing this is not known at this
> > +  # time (11 September 2015). This is not the only incident of its kind; see:
> > +  #     https://forums.developer.apple.com/thread/13795
> > +  # so it is possible that this can be removed if/when XCode and/or the CLT get
> > +  # fixed.
> > +  #
> > +  # Builds without these support libraries completed normally, which is why
> > +  # this change was not required earlier. In addition, the libraries are only
> > +  # built when clang is built
> > +
> > +  env :std if build.with? "clang"
> It'll only build 64-bit binaries unless universal is set.
> 
> —
> Reply to this email directly or view it on GitHub.
>